mcm/tools/installer.py

```diff
@@ -54,10 +52,77 @@ def close(self):
         if self.cleanup:
             try:
                 self.logger.error('Deleting the directory: %s' % self.directory)
-                shutil.rmtree(self.directory)
+                self._remove_folder(self.directory)
             except Exception as ex:
                 self.logger.error('Could not delete directory "%s". Reason: %s' % (self.directory, ex))
 
+    def _folder_exists(self, path):
+        """
+        Checks if a folder exists using the remote session.
+
+        Arguments:
+            path (str): Absolute path to check.
+
+        Returns:
+            bool: True if the folder exists, False otherwise.
+        """
+        _, stdout, stderr = self.ssh_session.execute("ls -l '%s'" % (path))
+        if not stdout and not stderr:
+            msg = "SSH error to check if the folder exists"
+            self.logger.error(msg)
+            raise RuntimeError(msg)
+        
+        output = stdout.read()
+        error = stderr.read()
+        if output and not error:
+            # Most probably we found a valid path
+            return True
+        if error and not output:
+            # Check the proper message is in the `stderr`
+            if "No such file or directory" in error:
+                return False
+            raise RuntimeError("Unexpected message for checking if a folder exists: %s" % (error))
+        
+    def _create_folder(self, path):
+        """
+        Create a folder using the remote session.
+
+        Arguments:
+            path (str): Absolute path to create.
+        """
+        _, stdout, stderr = self.ssh_session.execute("mkdir -p '%s'" % (path))
+        if not stdout and not stderr:
+            msg = "SSH error to create a folder: %s" % (path)
+            self.logger.error(msg)
+            raise RuntimeError(msg)
+        
+        error = stderr.read()
+        if error:
+            raise RuntimeError("Unable to create the folder: %s - Description: %s" % (path, error))
+
+        if not self._folder_exists(path):
+            raise RuntimeError("Folder not found (%s), it should exist" % (path))
+        
+    def _remove_folder(self, path):
```

Comment:
[rmdir()](https://docs.paramiko.org/en/latest/api/sftp.html#paramiko.sftp_client.SFTPClient.rmdir) on the SFTP session

Reply:
The folder will have some files inside, so just calling `rmdir()` will raise an exception here.

Reply:
ok fair enough